src/Infrastructure/Model/Shopware6Media.php

```diff
@@ -37,6 +47,100 @@ public function setFileName(string $fileName): void
 
     public function jsonSerialize(): array
     {
-        return ['fileName' => $this->fileName];
+        $data = [];
+        if (!is_null($this->fileName)) {
+            $data['fileName'] = $this->fileName;
+        }
+
+        foreach ($this->translations as $translation) {
+            $data['translations'][$translation->getLanguageId()] = $translation->jsonSerialize();
+        }
+
+        return $data;
+    }
+
+    public function isModified(): bool
+    {
+        return $this->modified;
+    }
+
+    public function setModified(): void
+    {
+        $this->modified = true;
+    }
+
+    /**
+     * @param Shopware6MediaTranslation[] $translations
+     */
+    public function setTranslations(array $translations): void
+    {
+        $this->translations = $translations;
+    }
+
+    /**
+     * @return Shopware6MediaTranslation[]
+     */
+    public function getTranslations(): array
+    {
+        return $this->translations;
+    }
+
+    /**
+     * @param Shopware6MediaTranslation[] $translations
+     */
+    public function updateTranslated(array $translations): void
+    {
+        $processedLanguages = [];
+        foreach ($translations as $translation) {
+            $processedLanguages[] = $translation->getLanguageId();
+            if (array_key_exists($translation->getLanguageId(), $this->translations)) {
+                if ($this->isTranslationChanged($translation, $this->translations[$translation->getLanguageId()])) {
+                    $this->translations[$translation->getLanguageId()] = new Shopware6MediaTranslation(
+                        $this->translations[$translation->getLanguageId()]->getId(),
+                        $translation->getAlt() ?? null,
+                        $translation->getTitle() ?? null,
+                        $translation->getLanguageId()
+                    );
+                    $this->setModified();
+                }
+                continue;
+            }
+            $this->translations[$translation->getLanguageId()] = new Shopware6MediaTranslation(
+                null,
+                $translation->getAlt() ?? null,
+                $translation->getTitle() ?? null,
+                $translation->getLanguageId()
+            );
+            $this->setModified();
+        }
```

Comment:
```suggestion
        foreach ($translations as $translation) {
            $processedLanguages[] = $translation->getLanguageId();
            if (array_key_exists($translation->getLanguageId(), $this->translations)) {
                continue;
            }
            
            if ($this->isTranslationChanged($translation, $this->translations[$translation->getLanguageId()])) {
                $id = $this->translations[$translation->getLanguageId()]->getId();
            }
            
            $this->translations[$translation->getLanguageId()] = new Shopware6MediaTranslation(
                $id ?? null,
                $translation->getAlt() ?? null,
                $translation->getTitle() ?? null,
                $translation->getLanguageId()
            );
            $this->setModified();
        }
```

Reply:
@filip we cannot continue until check if isTranslationChanged (found one) we cannot check on not found translations.

Reply:
condition reverted for better visibility